app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -1127,13 +1155,17 @@ public void handleResult(@NonNull final StreamInfo info) {
         animateView(thumbnailPlayButton, true, 200);
         videoTitleTextView.setText(name);
 
-        if (!TextUtils.isEmpty(info.getUploaderName())) {
-            uploaderTextView.setText(info.getUploaderName());
-            uploaderTextView.setVisibility(View.VISIBLE);
-            uploaderTextView.setSelected(true);
+        if (!TextUtils.isEmpty(info.getSubChannelName())) {
+            displayBothUploaderAndSubChannel(info);
+            subChannelThumb.setVisibility(View.VISIBLE);
+        } else if (!TextUtils.isEmpty(info.getUploaderName())) {
+            displayUploaderAsSubChannel(info);
         } else {
             uploaderTextView.setVisibility(View.GONE);
+            uploaderThumb.setVisibility(View.GONE);
```

Comment:
Shouldn't you also hide subChannelThumb? just to be sure that it is gone, because I think it could happen (i.e. when restoring instance) that `handleResult` is called twice and with different results, thus possibly resulting in wrongly showing the thumbnail left behind from the previous call. Not sure though if this is really an issue, but this is already the implemented behaviour with uploaderThumb

Reply:
I didn't do that, as it's gone by default in the .xml, so I thought that's not needed.

Reply:
Mmmh ok, it not the best to have hardcoded default values for things that could possibly be used multiple times, but maybe we can ignore it since if we were to make videodetailfragment reusable it would mean changing many things, so I guess this is just not important